gradle-server/src/main/java/com/github/badsyntax/gradle/GradleServer.java

```diff
@@ -42,24 +44,48 @@ public void run() {
 	}
 
 	public void stop() throws InterruptedException {
-		if (server != null) {
-			server.shutdown().awaitTermination(30, TimeUnit.SECONDS);
+		if (taskServer != null) {
+			taskServer.shutdown().awaitTermination(30, TimeUnit.SECONDS);
 		}
 	}
 
 	private void blockUntilShutdown() throws InterruptedException {
-		if (server != null) {
-			server.awaitTermination();
+		if (taskServer != null) {
+			taskServer.awaitTermination();
 		}
 	}
 
 	public static void main(String[] args) throws Exception {
-		int port = 8887;
-		if (args.length > 0) {
-			port = Integer.parseInt(args[0]);
+		Map<String, String> params = Utils.parseArgs(args);
+
+		int taskServerPort = Integer.parseInt(Utils.validateRequiredParam(params, "port"));
+		boolean startBuildServer = Boolean.parseBoolean(Utils.validateRequiredParam(params, "startBuildServer"));
+
+		startTaskServerThread(taskServerPort);
```

Comment:
Move this line after `int taskServerPort = Integer.parseInt(Utils.validateRequiredParam(params, "port"));`